db/indexes.go

```diff
@@ -185,13 +186,25 @@ func (i *SGIndex) createIfNeeded(bucket *base.CouchbaseBucketGoCB, useXattrs boo
 		return false, metaErr
 	}
 
-	// For already existing indexes, check whether they are deferred
+	// For already existing indexes, check whether they need to be built.
 	if exists {
 		if indexMeta == nil {
 			return false, fmt.Errorf("No metadata retrieved for existing index %s", indexName)
 		}
-		if indexMeta.State == base.IndexStateDeferred || indexMeta.State == base.IndexStatePending {
-			return true, nil
+		if indexMeta.State == base.IndexStateDeferred {
+			// Two possible scenarios when index already exists in deferred state:
+			//  1. Another SG is in the process of index creation
+			//  2. SG previously crashed between index creation and index build.
+			// GSI doesn't like concurrent build requests, so wait and recheck index state before treating as option 2
```

Comment:
"Doesn't like" is a bit vague .. could this instead say "GSI has a known issue w/ concurrent build requests", and then either give a brief summary of what happens or link to the version of the docs that describes that issue?

Reply:
Or better yet, link to the JIRA ticket -- that way it would be easy to lookup and check to see if it's possible to remove the workaround.

Reply:
That *would* be nice, if there was an open JIRA issue.

Reply:
Actually, can you point to `https://developer.couchbase.com/documentation/server/5.1/n1ql/n1ql-language-reference/build-index.html` rather than `../current/..`?

Reply:
Or `https://developer.couchbase.com/documentation/server/5.5/n1ql/n1ql-language-reference/build-index.html`